File: ze-app/src/main/java/com/example/zeapp/models/Word.java
package com.example.zeapp.models;


import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.Table;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "word")
public class Word {
    @Id
    @Column("id")
    private int id;
    @Column("foreign_word")
    private String foreignWord;
    @Column("transcription")
    private String transcription;
    @Column("translation")
    private String translation;
    @Column("description")
    private String description;
    @Column("link_voice")
    private String link_voice;
    @Column("link_image")
    private String link_image;
    @Column("topic")
    private String topic;
    @Column("sorting_value")
    private Integer sorting_value;
}

File: ze_admin(android)/app/src/main/java/com/example/ze_adminandroid/models/Word.kt
package com.example.ze_adminandroid.models

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey


@Entity(tableName = "editable_words")
data class Word(

    //это поле для адекватной работы с местной базой данных конкретно в этом приложении,
    // при этом строго запрещено ей менять id приходящее с сервера,
    // и новые созданные Word отправляются на сервер с значением id = null
    @PrimaryKey
    val localBaseId: Int?,


    @ColumnInfo(name = "id")
    val id: Int,
    @ColumnInfo(name = "foreignWord")
    val foreignWord:String?,
    @ColumnInfo(name = "transcription")
    val transcription:String?,
    @ColumnInfo(name = "translation")
    val translation:String?,
    @ColumnInfo(name = "description")
    val description:String?,
    @ColumnInfo(name = "topic")
    val topic:String,
    @ColumnInfo(name = "link_voice")
    val link_voice: String?,
    @ColumnInfo(name = "link_image")
    val link_image: String?,
    @ColumnInfo(name = "sorting_value")
    val sorting_value: Int
)
